routers/web/repo/view.go

```diff
@@ -495,6 +495,11 @@ func renderFile(ctx *context.Context, entry *git.TreeEntry, treeLink, rawLink st
 		ctx.Data["ReadmeExist"] = readmeExist
 
 		markupType := markup.Type(blob.Name())
+		detected := false
+		if markupType == "" {
+			detected = true
```

Comment:
I think your original logic is wrong:
```suggestion
		detected := markupType != ""
		if !detected {
```

Reply:
no it's not wrong it's meant to reset markupType only if automatic detection was used on L512, with your proposed solution it would do it other way around

Reply:
Yes. I know that I turned the boolean around.
Because at the moment it simply looks wrong to me when reading the code - a boolean `detected` checking if the content type has not been detected is wrong for me.

Reply:
because markupType needs to be returned to view (`ctx.Data["MarkupType"] = markupType`) but should not be passed down to `markup.Render` func in `markup.RenderContext` struct